### PR TITLE
User Model and Tests

### PR DESCRIPTION
**Pull request by ttakahashi1591**
## Description
This PR includes the following features and test:

- Builds out the `User` Model for all `CRUD` functioning
- Tests for all `User` model `CRUD` functions
- Set up all necessary gems  

## Type of Change

- [ ] **fix**
- [x] **feat**
- [x] **test**
- [ ] **refactor**
- [ ] **docs**

## Checklist

- [x] **Code has been self reviewed**
- [x] **Code runs without any errors**
- [x] **Thorough testing has been implemented if adding feature**
- [x] **All tests pass**

### CODE REVIEW

File: app/models/user.rb
Comment (reedhillmar) [inline app/models/user.rb:1]: we'll need to add relationships once parties are a thing

File: pull_request_template.md
Comment (reedhillmar) [inline pull_request_template.md:1]: I love a good template

File: spec/models/user_spec.rb
Comment (reedhillmar) [inline spec/models/user_spec.rb:1]: We'll need to test validations

File: spec/requests /api/v1/users_request_spec.rb
Comment (reedhillmar) [inline spec/requests /api/v1/users_request_spec.rb:1]: Yay for using resources and paths!

**Review by reedhillmar — APPROVED** [review]
We pair programmed through this whole thing. Great job, TT!

File: app/models/user.rb
Comment (ttakahashi1591) [reply app/models/user.rb:1]: Exactly what I am thinking!

File: spec/models/user_spec.rb
Comment (ttakahashi1591) [reply spec/models/user_spec.rb:1]: We can work on building model specs for both Users and Parties

**Pull request merged**